flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -513,12 +509,8 @@ private void validateClusterSpecification(ClusterSpecification clusterSpecificat
 			yarnApplication,
 			validClusterSpecification);
 
-		String host = report.getHost();
-		int port = report.getRpcPort();
-
-		// Correctly initialize the Flink config
-		flinkConfiguration.setString(JobManagerOptions.ADDRESS, host);
-		flinkConfiguration.setInteger(JobManagerOptions.PORT, port);
```

Comment:
I'm not 100% sure that we can remove this update operation. A quick search revealed that the `ResultStore` uses the `JobManagerOptions.ADDRESS`. Since I could not rule out that we implicitly rely on the fact that the `flinkConfiguration` is shared, I would not remove this setting. Only if we are sure, ideally guarded by a test, then we should remove it.

Reply:
This configuration is passed to `RestClusterClient` and never pass to `ResultStore` IIUC.

An investigation of generally how we configure and use JM(DispatcherRMComponent) and WebMonitor host&port would be a worthy follow-up. I don't stick to remove this since it does no harm in production but given its outputs are only when constructs `RestClusterClient` in `createYarnClusterClient` we can ensure that it would not escape.

Reply:
Aha! Even I found a reason that we "need" these stmts because when build HAService at client-side we have to build one just like on cluster client thus if `JobManagerOption` not set it will fail with

```
org.apache.flink.client.deployment.ClusterDeploymentException: Could not deploy Yarn job cluster.
	at org.apache.flink.yarn.YARNITCase.lambda$testPerJobMode$0(YARNITCase.java:109)
	at org.apache.flink.yarn.YARNITCase.testPerJobMode(YARNITCase.java:69)
Caused by: org.apache.flink.util.ConfigurationException: Config parameter 'Key: 'jobmanager.rpc.address' , default: null (fallback keys: [])' is missing (hostname/address of JobManager to connect to).
	at org.apache.flink.yarn.YARNITCase.lambda$testPerJobMode$0(YARNITCase.java:109)
	at org.apache.flink.yarn.YARNITCase.testPerJobMode(YARNITCase.java:69)
```

this is what FLINK-13750 want to resolve, i.e., do not instance service that is never needed.

Reply:
Reverted removals and create FLINK-13929 for further discussion.